transmogrifier/sources/ead.py

```diff
@@ -452,3 +424,59 @@ def parse_mixed_value(
         elif isinstance(item, Tag) and item.name not in skipped_elements:
             for child in item.children:
                 yield from cls.parse_mixed_value(child, skipped_elements)
+
+    def parse_dates(
```

Comment:
Good call pulling this into a new method, it's too involved not to be separated out from `get_optional_fields`